contracts/src/main/kotlin/com/template/contracts/AppointmentRequestContract.kt

```diff
@@ -26,16 +25,56 @@ class AppointmentRequestContract : Contract {
     }
 
     override fun verify(tx: LedgerTransaction) {
-        requireThat {
-            "No inputs should be consumed when issuing a date" using (tx.inputs.isEmpty())
-            "1 reference state should be used" using (tx.references.size == 1)
-            "Only one output state is created" using (tx.outputs.size == 1)
+        val command = tx.findCommand<Commands> { true }
 
-            val out = tx.outputStates[0] as AppointmentRequest
-            "Dates must be of the format dd-MM-yyyy" using (checkDate(out.date))
+        when (command.value) {
+            is Commands.Create -> {
+                requireThat {
+                    "No inputs should be consumed when issuing a date" using (tx.inputs.isEmpty())
+                    "1 reference state should be used" using (tx.references.size == 1)
+                    "Only one output state is created" using (tx.outputs.size == 1)
+
+                    val ref = tx.referenceInputRefsOfType<AvailableAppointmentDate>()[0].state.data
+                    "Date must match format dd-MM-yyyy" using (checkDate(ref.date))
+                }
+            }
+            is Commands.Accept -> {
+                requireThat {
+                    "2 inputs should be consumed when accepting a request" using (tx.inputs.size == 2)
+                    "No reference states should be used when accepting a request" using (tx.references.isEmpty())
+                    "1 Output state is created" using (tx.outputs.size == 1)
+
+                    val in1 = tx.inputsOfType<AvailableAppointmentDate>()[0]
+                    val in2 = tx.inputsOfType<AppointmentRequest>()[0]
+
+                    "The available date and requested date must be the same" using (in1.date == in2.date)
+
+                    val signer = tx.commandsOfType<Commands.Accept>()[0].signers[0]
+                    val doctor = in1.doctor
+
+                    "The doctor must be the first signer of the transaction" using (signer == doctor.owningKey)
```

Comment:
It doesn't really matter in this case but it's better to check whether a doctor is one of signers, not necessarily the first one.

Reply:
Shouldn't the doctor be the only signer?